### ISSUE TITLE
Bug: "plugin:@typescript-eslint/recommended" error eslint JS files

### ISSUE THREAD

**Issue by Colemanhawkins** [typescript-eslint/typescript-eslint#8395]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

ast-spec

### Repro Code

Hi there. I always appreciate the work you guys do.

I tried to run lint with @typescript-eslint/parser as a parser.

and works but I am having the problem that the linter to all my js files either babel.config.js or any js I create in the seed folder or inside any folder, eslint gives me at the beginning of the files the following error

error Parsing error: '{' expected.eslint

I would appreciate it if you could tell me the cause and solution.

me repo url is : https://github.com/monkeywit/nodejs-typescript-setup

i show mi ts.config.json:
```json
{
  "compilerOptions": {
    "target": "es2016",
    "module": "commonjs",                                
    "rootDir": "./src",
    "allowJs": true,                                  
    "outDir": "build",
    "baseUrl": ".",
    "paths": {
      "@routes/*": ["./src/routes/*"],
      "@constrollers/*": ["./src/controllers/*"],
      "@config/*": ["./src/confing/*"]
    },
    "typeRoots": ["./node_modules/@types", "./src/@types"],   
    "esModuleInterop": true,                         
    "forceConsistentCasingInFileNames": true,                     
    "skipLibCheck": true ,
    "strictNullChecks": true                               
  },
  "include": ["src/**/*"],
  "exclude": ["node_modules"]
}
```

eslintrc.json

```json
{
  "env": {
    "es2021": true,
    "node": true
  },
  "extends": [
    "standard-with-typescript",
    "plugin:prettier/recommended",
    "eslint:recommended",
    "plugin:@typescript-eslint/recommended"
  ],
  "plugins": ["@typescript-eslint", "import", "prettier"],
  "overrides": [],
  "parserOptions": {
    "ecmaVersion": "latest",
    "sourceType": "module",
    "project": "./tsconfig.json"
  },
  "rules": {
    "prettier/prettier": "error",
    "import/no-unresolved": [
      "error",
      {
        "plugins": [
          "module-resolve",
          {
            "alias": {
              "@routes": "./src/routes",
              "@constrollers": "./src/controllers",
              "@config": "./src/confing"
            }
          }
        ]
      }
    ]
  }
}
```

and version of me dependences: 

```json
{
  "name": "typescript-nodejs-setup",
  "version": "1.0.0",
  "main": "index.js",
  "author": "Efrain Rodriguez <efrainrg92@gmail.com>",
  "license": "MIT",
  "scripts": {
    "build": "babel src --extensions \".ts,.js\" --out-dir build --copy-files",
    "start": "node build/index.js",
    "dev": "ts-node-dev -r tsconfig-paths/register --respawn --ignore-watch node_modules src/index.ts"
  },
  "devDependencies": {
    "@babel/cli": "^7.18.10",
    "@babel/core": "^7.19.0",
    "@babel/node": "^7.18.10",
    "@babel/preset-env": "^7.19.0",
    "@babel/preset-typescript": "^7.18.6",
    "@types/express": "^4.17.13",
    "@typescript-eslint/eslint-plugin": "^5.0.0",
    "babel-plugin-module-resolver": "^4.1.0",
    "eslint": "^8.0.1",
    "eslint-config-prettier": "^8.5.0",
    "eslint-config-standard-with-typescript": "^22.0.0",
    "eslint-import-resolver-alias": "^1.1.2",
    "eslint-plugin-import": "^2.25.2",
    "eslint-plugin-n": "^15.0.0",
    "eslint-plugin-prettier": "^4.2.1",
    "eslint-plugin-promise": "^6.0.0",
    "prettier": "^2.7.1",
    "ts-node-dev": "^2.0.0",
    "tsconfig-paths": "^4.1.0",
    "typescript": "*"
  },
  "dependencies": {
    "eslint-import-resolver-typescript": "^3.6.1",
    "express": "^4.18.1"
  }
}
```

### ESLint Config

```javascript
{
  "env": {
    "es2021": true,
    "node": true
  },
  "settings": {
    "import/resolver": {
      "alias": {
        "map": [
          ["@routes", "./src/routes"],
          ["@controllers", "./src/controllers"],
          ["@config", "./src/config"]
          // Agrega otros alias según sea necesario
        ],
        "extensions": [".ts", ".tsx", ".js", ".jsx"]
      }
    }
  },
  "extends": [
    "standard-with-typescript",
    "plugin:prettier/recommended",
    "eslint:recommended",
    "plugin:@typescript-eslint/recommended"
  ],
  "plugins": ["@typescript-eslint", "import", "prettier"],
  "overrides": [],
  "parserOptions": {
    "ecmaVersion": 2021,
    "sourceType": "module",
    "project": ["./tsconfig.json", "./babel.config.js"]
  },
  "rules": {}
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "target": "es2021",
    "module": "commonjs",
    "rootDir": "./src",
    "allowJs": true,
    "outDir": "build",
    "baseUrl": ".",
    "paths": {
      "@routes/*": ["./src/routes/*"],
      "@constrollers/*": ["./src/controllers/*"],
      "@config/*": ["./src/confing/*"]
    },
    "typeRoots": ["./node_modules/@types", "./src/@types"],
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "forceConsistentCasingInFileNames": true,
    "skipLibCheck": true,
    "strictNullChecks": true
  },
  "include": ["src/**/*"],
  "exclude": ["node_modules"]
}
```

### Expected Result

eslint with not problems 

### Actual Result

eslint dont show me parser error in me JS files , I know it is the "plugin:@typescript-eslint/recommended" configuration that is the problem because when I remove it from my eslintrc.json the js files stop showing me error eslint for the indentation error.

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `5.0.0` |
| `@typescript-eslint/parser`            | `X.Y.Z` |
| `@typescript-eslint/rule-tester`       | `X.Y.Z` |
| `@typescript-eslint/scope-manager`     | `X.Y.Z` |
| `@typescript-eslint/typescript-estree` | `X.Y.Z` |
| `@typescript-eslint/type-utils`        | `X.Y.Z` |
| `@typescript-eslint/utils`             | `X.Y.Z` |
| `TypeScript`                           | `5.3.3` |
| `ESLint`                               | `8.0.1` |
| `node`                                 | `18.8.0` |

**Comment by bradzacher** [comment]
Have you read our getting started guide? I would suggest going back to basics and following our guide to help you debug your configuration.

https://typescript-eslint.io/getting-started

Cut your config down to zeor and rebuild it using our guide and you'll be able to pinpoint where things might have gone wrong.

**Comment by bradzacher** [comment]
In general issues are for *bug* reports. If you're looking for support - as per our contributing guide - there are much better places to ask:

https://typescript-eslint.io/contributing/issues#questions-and-support-requests
